climpyrical/cmd/preprocess_model.py

```diff
@@ -0,0 +1,225 @@
+
+    nanmask = ~np.isnan(mean)
+    points = np.stack([rlon[nanmask], rlat[nanmask]]).T
+    target_values = mean[nanmask]
+    target_points = np.stack([rlon[glaciermask], rlat[glaciermask]]).T
+
+    # interpolate NaN values using bilinear interpolation
+    mean[glaciermask] = interpolate_dataset(
+        points, target_values, target_points, "linear"
+    )
+    ds = gen_dataset(dv, mean, ds.rlat, ds.rlon, ds.lat, ds.lon)
+
+    # nanmask = ~np.isnan(mean)
+    ds = ds.assign({dv: (["rlat", "rlon"], mean)})
+
+    # make all water values NaN at original resolution
+    ds[dv].values[~mask_og] = np.nan
+    nanmask = ~np.isnan(ds[dv].values)
+
+    # copy newly masked dv field to target resolution
+    ds10 = regrid_ensemble(ds, dv, 10, copy=True)
+
+    # Mask out ocean values
+    ds10[dv].values[~mask] = np.nan
+    nrlon, nrlat = np.meshgrid(ds10.rlon, ds10.rlat)
+    nanmask10 = ~np.isnan(ds10[dv].values)
+
+    print("Interpolating full grid.")
+
+    # bilinearly interpolate over non NaN grids
+    points = np.stack([rlon[nanmask], rlat[nanmask]]).T
+    target_points = np.stack([nrlon[nanmask10], nrlat[nanmask10]]).T
+    values = ds[dv].values[nanmask]
+    ds10[dv].values[nanmask10] = interpolate_dataset(
+        points, values, target_points, "linear"
+    )
+
+    # extend the northern region
+    ds10 = extend_north(ds10, dv, 210, fill_val=np.nan)
+
+    # create nanmask for interpolation
+    nanmask10 = ~np.isnan(ds10[dv].values)
+
+    # load processed canada-only land mask
+    canada_mask_path = resource_filename(
+        "climpyrical", "/tests/data/canada_mask_rp.nc"
+    )
+
+    with read_data(canada_mask_path) as ds_canada:
+        ca_mask = extend_north(ds_canada, "mask", 210, fill_val=np.nan)
+        ca_mask = ds_canada["mask"].values
+
+    # select NaN values within new mask
+    ca_mask_or = ~np.logical_or(~ca_mask, nanmask10)
+
+    # Fill inconsistent points using closest neighbour.
+    # Not that the northern section will be filled later on
+    nrlon, nrlat = np.meshgrid(ds10.rlon.values, ds10.rlat.values)
+
+    temp_field = ds10[dv].values
+
+    points = np.stack([nrlon[nanmask10], nrlat[nanmask10]]).T
+    target_points = np.stack([nrlon[ca_mask_or], nrlat[ca_mask_or]]).T
+    target_values = ds10[dv].values[nanmask10]
+    temp_field[~ca_mask] = np.nan
+
+    print("Detecting missing points from canada mask and filling.")
+
+    temp_field[ca_mask_or] = interpolate_dataset(
+        points, target_values, target_points, "nearest"
+    )
+
+    # UAA "unfill"
+    uaa_mask_path = resource_filename(
+        "climpyrical", "tests/data/canada_mask_north_rp.nc"
+    )
+
+    uaa_mask = read_data(uaa_mask_path)["mask"]
+    temp_field[uaa_mask] = np.nan
+
+    # create final dataset
+    ds_processed = gen_dataset(
+        dv, temp_field, ds10.rlat, ds10.rlon, ds10.lat, ds10.lon
+    )
+
+    print("Dataset generated and writing to file.")
+
+    if os.path.exists(OUT_PATH):
+        os.remove(OUT_PATH)
```

Comment:
This seems a bit dangerous. We _do_ check for a `.nc` suffix before getting into this method but we can still call this method directly if we wanted to (in testing for example). I would recommend moving the arg check into this function so that no accidents happen.

Why are we removing the existing `.nc` file in the first place? Wouldn't it make sense to overwrite the result or throw an error stating the name is already taken?

Reply:
The intention here _is_ to overwrite the result.

You're right though, simply using:
```
ds_processed.to_netcdf(out_path, "w")
```

[Should overwrite the existing file.](https://xarray.pydata.org/en/v0.16.1/generated/xarray.Dataset.to_netcdf.html#xarray.Dataset.to_netcdf)